api/pkg/controller/seed-proxy/reconciler.go

```diff
@@ -337,3 +340,21 @@ func (r *Reconciler) reconcileMasterGrafanaProvisioning(seeds map[string]*kuberm
 
 	return nil
 }
+
+func (r *Reconciler) deleteResource(client ctrlruntimeclient.Client, name string, namespace string, obj runtime.Object) error {
+	key := types.NamespacedName{Name: name, Namespace: namespace}
+
+	if err := client.Get(context.Background(), key, obj); err != nil {
```

Comment:
tiny notice: probably should be
```suggestion
	if err := client.Get(r.ctx, key, obj); err != nil {
```